### DISCUSSION TITLE
Auto mouse layer の公式サポートについて

### DISCUSSION THREAD

**Discussion by tokuhirom** [Yowkees/keyball#449] [Ideas]
Firmware v1.2.0 で、QMK 0.22.x based になったので、　https://github.com/Yowkees/keyball/discussions/　で言及されている、QMK の auto mouse layer 機能が使えるようになったと思います。
インターネットを見ている感じ、auto mouse layer を採用している人も多いので、公式レポジトリでのファームウェアでも hex ファイルを提供されてると便利なのかなぁと思いました。

(あるいはドキュメントに追記するだけでも助かる人が多い気がしました)

↓私自身は、keymaps/via をベースに以下のようにして使っています

```
diff -u via/config.h auto-mouse-layer/config.h
--- via/config.h	2023-12-15 12:17:31
+++ auto-mouse-layer/config.h	2023-12-15 13:39:06
@@ -33,4 +33,8 @@
 //#    define RGBLIGHT_EFFECT_TWINKLE
 #endif
 
+#define POINTING_DEVICE_AUTO_MOUSE_ENABLE
+// Time layer remains active after activation
+#define AUTO_MOUSE_TIME 450
+
 #define TAP_CODE_DELAY 5
diff -u via/keymap.c auto-mouse-layer/keymap.c
--- via/keymap.c	2023-12-15 12:17:31
+++ auto-mouse-layer/keymap.c	2023-12-15 13:05:33
@@ -72,3 +72,8 @@
     keyball_oled_render_layerinfo();
 }
 #endif
+
+void pointing_device_init_user(void) {
+    set_auto_mouse_layer(2);
+    set_auto_mouse_enable(true);
+}
```

**Comment by Yowkees** [comment]
Automatic mouse layer機能はトラックボールを操作した時に特定のレイヤーに切り替わる機能ですね！
一時期気になっていたのですが失念していました。

公式のhexファイルに加えられるように検討しますので少々お待ちください。

ご指摘ありがとうございました。

**Reply by tokuhirom** [reply]
https://github.com/Yowkees/keyball/pull/454 実装できそうだったので、実装してみました。
